charts/opentelemetry-collector/templates/daemonset.yaml

```
@@ -10,6 +10,19 @@ spec:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
       component: agent-collector
+  updateStrategy:
+    {{- if not (eq .Values.agentCollector.updateStrategy "OnDelete") }}
```

```suggestion
    {{- if eq .Values.agentCollector.updateStrategy "RollingUpdate" }}
```
Since this section is only valid when the updateStrategy is `RollingUpdate`, lets explicitly check for it to be set (even tho its the default).

---

Agreed. Also `deployment` mode will bring another option after https://github.com/open-telemetry/opentelemetry-helm-charts/pull/159. So we need to be precise here